File: Golang Solutions/Counting Bits.go
func countBits(num int) []int {
    var vec []int
    var tmp []int
    vec=append(vec,0);
    for i:=1;i<=num;{
        for _,x:=range vec{
            tmp=append(tmp,x+1)
            i++
            if i>num{
                break
            }
        }
        vec=append(vec,tmp...)
        tmp=nil
    }
    return vec
}

File: Counting Bits.cpp
class Solution {
public:
   
    vector<int> countBits(int num) {
        vector<int>vec;
        vector<int>tmp;
        vec.push_back(0);
        for(int i=1;i<=num;)
        {
            for(auto x: vec)
            {
                tmp.push_back(x+1);
                i++;
                if(i>num)
                    break;
            }
            vec.insert(vec.end(),tmp.begin(),tmp.end());
            tmp.clear();
        }
        return vec;
        
    }
};
